### DISCUSSION TITLE
make workflow.json output action ordering more deterministic

### DISCUSSION THREAD

**Discussion by thinkOfaNumber** [Azure/logicapps#956] [Feature Ideas/Requests]
I noticed somewhere around version v2.86.5 of Azure Logic Apps (Standard), the ordering of properties for an action has become alphabetical. This has only been a slight problem when multiple developers have different versions, but since it's more deterministic, it's better for code analysis (github, pull requests, blame, comparing revisions across branches, etc).

Similarly, I think the entire workflow.json output ordering should be more deterministic. At the moment, it seems actions are placed in the order that they are created, so if I create a workflow with action dependencies like this:
```trigger > action 1 > action 2 > action 3```
the workflow.json has them in that order. Then, if I add an action between 1 and 2:
```trigger > action 1 > action 1b > action 2 > action 3```
the workflow.json shows them in this order:
```trigger, action 1, action 2, action 3, action 1b ```
and uses the `runAfter` property to display them in the correct order.

The problems with this are not limited to:

1. creating an identical-looking logic app may end up with different workflow.json
2. deleting and adding the same action may end up with different workflow.json
3. as a result of the above, comparing differences between branches/commits becomes useless, as the workflow.json ordering can change even when there is no functional difference to the logic app

I would like to see the workflow.json reordered on every save to be deterministic, perhaps relevant to the runAfter exectution order.

**Comment by mark-abrams** [comment]
I completely agree with this suggestion. The ordering should be deterministic to make it easier to compare versions and review PRs. 

I remember that the portal based workflow designer and the local designer in VS Code used to order actions differently which was a real pain when reviewing, I'm not sure if this is still an issue though.
